### PR TITLE
Multiple tables

### PR DESCRIPTION
**Pull request by AdnanOlarmmi**
# Vet clinic database: query multiple tables

Hi Reviewer, 
Patrick Hi

## Learning objectives
- Use primary key & foreign key mechanism for joining tables.
- Query multiple tables.
- Prepare complex queries that answer analytical questions.

## Description
In this project, I added some new tables and added foreign key columns to my existing `animals` table to model **one-to-many** relationships. Afterward, I used `JOIN` to query the data.

### General requirements

- Made sure that I used correct ([Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)).
- Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### Project requirements
- Created a table named `owners` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `full_name`: string
  - `age`: integer
- Created a table named `species` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
- Modified `animals` table:
  - Made sure that id is set as autoincremented PRIMARY KEY
  - Removed column `species`
  - Added column `species_id` which is a foreign key referencing `species` table
  - Added column `owner_id` which is a foreign key referencing the `owners` table
- Remembered all this goes in `schema.sql` file.
- Inserted the following data into the `owners` table:
  - Sam Smith 34 years old.
  - Jennifer Orwell 19 years old.
  - Bob 45 years old.
  - Melody Pond 77 years old.
  - Dean Winchester 14 years old.
  - Jodie Whittaker 38 years old.
- Inserted the following data into the `species` table:
  - Pokemon
  - Digimon
- Modified my inserted animals so it includes the `species_id` value:
  - If the name ends in "mon" it will be Digimon
  - All other animals are Pokemon
- Modified my inserted animals to include owner information (`owner_id`):
  - Sam Smith owns Agumon.
  - Jennifer Orwell owns Gabumon and Pikachu.
  - Bob owns Devimon and Plantmon.
  - Melody Pond owns Charmander, Squirtle, and Blossom.
  - Dean Winchester owns Angemon and Boarmon.
- Remembered these insertions and modifications happen in `data.sql` file.

- Wrote queries (using `JOIN`) to answer the following questions:
  - What animals belong to Melody Pond?

  - List of all animals that are pokemon (their type is Pokemon).

  - List all owners and their animals, remember to include those that don't own any animal.

  - How many animals are there per species?

  - List all Digimon owned by Jennifer Orwell.

  - List all animals owned by Dean Winchester that haven't tried to escape.

  - Who owns the most animals?

- Remembered all these should be written in `queries.sql` file.
- **Took a screenshot of the results of my queries.**
- Remembered to add the screenshot with the results of my queries to your Pull Request description.

### Need a big picture?

Remind me about [the big picture of this project](./sneak_peek.md).

## Work and submission mode

- Submitted **individually.**

## Appreciation
- Thanks for the review, it means a lot 🙃

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -45,7 +45,20 @@ SELECT SUM(escape_attempts) AS escape_attempts, neutered FROM animals GROUP BY n
 SELECT MAX(weight_kg), MIN(weight_kg), species FROM animals GROUP BY species;
 SELECT AVG(escape_attempts) AS avg_escape_attempts, species FROM animals  WHERE date_of_birth BETWEEN '1990/01/01' AND '2020/12/31' GROUP BY species;
 
-
+--What animals belong to Melody Pond?
+SELECT * FROM owners JOIN animals ON animals.owner_id = owners.id WHERE owners.full_name = 'Melody Pond';
```
Comment (Deepakdanger) [inline queries.sql]: - [x] Here you have written incorrectly we have to show `which animals belong to Melody Pond?`
```
SELECT * FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   WHERE full_name = 'Melody Pond';
```

File: queries.sql [hunk 2]
```diff
@@ -45,7 +45,20 @@ SELECT SUM(escape_attempts) AS escape_attempts, neutered FROM animals GROUP BY n
...
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT * FROM animals JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT * FROM Owners FULL OUTER JOIN animals ON owners.id = animals.owner_id;
+-- How many animals are there per species?
+SELECT species.name, COUNT(*) FROM species JOIN  animals ON animals.species_id = species.id GROUP BY species.name;
```
Comment (Deepakdanger) [inline queries.sql]: - Here also 
```sql
 SELECT species.name, COUNT(*) FROM animals
   LEFT JOIN species
   ON animals.species_id = species.id
   GROUP BY species.name
```

File: queries.sql [hunk 3]
```diff
@@ -45,7 +45,20 @@ SELECT SUM(escape_attempts) AS escape_attempts, neutered FROM animals GROUP BY n
...
+-- List all Digimon owned by Jennifer Orwell.
+SELECT * FROM animals JOIN owners ON owners.id = animals.owner_id JOIN species ON species.id = animals.species_id WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT * FROM animals JOIN owners ON animals.owner_id = owners.id WHERE animals.escape_attempts = 0 AND owners.full_name = 'Dean Winchester';
+--Who owns the most animals?
+SELECT owners.full_name, COUNT(*) FROM animals JOIN owners ON animals.owner_id = owners.id GROUP BY owners.full_name ORDER BY COUNT DESC LIMIT 1;
```
Comment (Deepakdanger) [inline queries.sql]: - [x] You have written incorrectly. It should be 
`SELECT owners.full_name, COUNT(*) FROM animals JOIN owners ON `animals.speceis_id` = owners.id GROUP BY owners.full_name ORDER BY COUNT DESC LIMIT 1;`

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @AdnanOlarmmi , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (AdnanOlarmmi) [reply queries.sql]: Thanks @Deepakdanger , it was a sort of misunderstanding.

**Review by uwadonat — APPROVED** [review]
## Hi @AdnanOlarmmi,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
### Highlights
- Descriptive pull request message ✔️
- Good communication with reviewers ✔️
awoseme



# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**